api/storage_service.cc

```diff
@@ -1499,6 +1499,7 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto token = dht::token::from_int64(validate_int(req->get_query_param("token")));
         auto ks = req->get_query_param("ks");
         auto table = req->get_query_param("table");
+        validate_table(ctx, ks, table);
```

Comment:
Why is find_column_family() calling on_internal_error()? It seems like a normal condition for a table to be dropped.
\cc @bhalevy

Reply:
sorry for the confusion.

we typically call `find_uuid()` first. which throws `no_such_column_family`. we are able to catch this exception. and it does not abort.  and then look up the column_family table with the `table_id`, but if this step fails to find the table, it calls `on_internal_error()`. as it is a signal of a logical problem in our code. see https://github.com/scylladb/scylladb/blob/876478b84fd0f3a2e67e9385c562c9988973268a/replica/database.cc#L1151-L1158 .

i think it would be nice to return a better exception than `no_such_column_family`, hence this change.

Reply:
right.  if the table is not found it will throw no_such_column_family.  The internal error happens only if we find the table's id but not a corresponding object using this uuid.

Reply:
> i think it would be nice to return a better exception than `no_such_column_family`

What's wrong with `no_such_column_family`?

Reply:
@bhalevy hi Benny, 2 reasons:

1. `no_such_column_family` is handled by https://github.com/scylladb/seastar/blob/b83d298786a8984a8aea14bd3a273b46014642be/src/http/routes.cc#L75-L83, and is translated to an HTTP 500 error, but it's wrong. we should not blame the server for the bad request. if we really have 500 in our access log, we should be alerted as it is an indication of server-side problem. but in this case, the problem reside on the client-side.
2. it is not consistent with other places where we use `validate_table()`. in incurs the pain of the future developers -- one would ask, under which circumstances shall we throw `no_such_column_family`, and under which circumstances shall we throw `bad_param_exception` ?